x/genutil/client/cli/gentx.go

```diff
@@ -140,12 +140,7 @@ func GenTxCmd(ctx *server.Context, cdc *codec.Codec, mbm module.BasicManager, sm
 				return errors.Wrap(err, "failed to build create-validator message")
 			}
 
-			info, err := txBldr.Keybase().Get(name)
-			if err != nil {
-				return errors.Wrap(err, "failed to read from tx builder keybase")
-			}
-
-			if info.GetType() == kbkeys.TypeOffline || info.GetType() == kbkeys.TypeMulti {
+			if key.GetType() == kbkeys.TypeOffline || key.GetType() == kbkeys.TypeMulti {
```

Comment:
why don't you use 
```go
keyType := key.GetType()
if keyType == kbkeys.TypeOffline || keyType == kbkeys.TypeMulti {
...
}
```go

Reply:
Performance-wise it makes no difference, the value is cached anyway. Happy to change it if you think it's really worth it.

Reply:
which are the functions that require a password prompt?

Reply:
Every keyring initialization (e.g. `NewKeyringFile` calls) opens a new connection to the underlying backend. Thus, in the case of file-backed keyring, given that user session by design is not handled, all New calls will prompt users for password. Note that the `gentx` command is an exceptional case, this should not happen with any other command